app/setup_handlers.go

```diff
@@ -4,23 +4,40 @@ import (
 	storetypes "github.com/cosmos/cosmos-sdk/store/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	m "github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/cosmos/cosmos-sdk/x/authz"
+	"github.com/cosmos/cosmos-sdk/x/feegrant"
 	"github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	ibcconnectiontypes "github.com/cosmos/ibc-go/v2/modules/core/03-connection/types"
 )
 
 const upgradeName = "0.10.0-rc.1"
 
 func SetupHandlers(app *SifchainApp) {
 	app.UpgradeKeeper.SetUpgradeHandler(upgradeName, func(ctx sdk.Context, plan types.Plan, vm m.VersionMap) (m.VersionMap, error) {
 		app.Logger().Info("Running upgrade handler for " + upgradeName)
-		return vm, nil
+		app.IBCKeeper.ConnectionKeeper.SetParams(ctx, ibcconnectiontypes.DefaultParams())
+		fromVM := make(map[string]uint64)
+		// Old Modules can execute Migrations if needed .
+		// State migration logic should be added to a migrator function
+		// Migrating modules should increment the ConsensusVersion
+		for moduleName := range app.mm.Modules {
+			fromVM[moduleName] = 1
+		}
+		// New Modules must execute Init Genesis
+		fromVM[authz.ModuleName] = 0
+		fromVM[feegrant.ModuleName] = 0
+		fromVM["vesting"] = 0
```

Comment:
I don't see vesting or crisis installed as a new module?

Reply:
The current master does not have vesting or crisis module . I assumed that master is what is deployed on presently on betanet 
https://github.com/Sifchain/sifnode/blob/master/app/app.go

Or do you mean that 
We don't need to add these modules during our 0.44 upgrade ?  
I am actually not clear on why these modules were added , these were already present when I started working on this PR , I assumed we need these modules . Mayber @iverc can comment .

Reply:
@Jedi2002 I don't believe vesting is required. Any further thoughts?

Reply:
@intl-man @Jedi2002 @timlind Vesting module is not required, but could be useful in the near future. I'd include it to avoid having a separate upgrade for it specifically as long as it does not introduce issues

Reply:
That makes sense @iverc.